plugins/CoreHome/Tracker/VisitRequestProcessor.php

```diff
@@ -0,0 +1,208 @@
+use Piwik\Exception\UnexpectedWebsiteFoundException;
+use Piwik\Tracker\Cache;
+use Piwik\Tracker\Request;
+use Piwik\Tracker\RequestProcessor;
+use Piwik\Tracker\Settings;
+use Piwik\Tracker\Visit\VisitProperties;
+use Piwik\Tracker\VisitExcluded;
+use Piwik\Tracker\VisitorRecognizer;
+
+/**
+ * Encapsulates core tracking logic related to visits.
+ *
+ * ## Request Metadata
+ *
+ * This RequestProcessor exposes the following metadata for the **CoreHome** plugin:
+ *
+ * * **visitorId**: A hash that identifies the current visitor being tracked. This value is
+ *                  calculated using the Piwik\Tracker\Settings;:getConfigId() method.
+ *
+ *                  Set in `processRequestParams()`.
+ *
+ * * **isVisitorKnown**: True if the current visitor has visited the site before. False if
+ *                       otherwise.
+ *
+ *                       Set in `processRequestParams()`.
+ *
+ * * **isNewVisit**: True if the current action is the start of a new visit, false if it
+ *                   is part of an ongoing visit.
+ *
+ *                   Set in `processRequestParams()`. Other RequestProcessors can override
+ *                   this value to force a new visit or stop a new visit.
+ *
+ * * **visitorNotFoundInDb**: True if the current visit could not be updated.
+ *
+ *                            Set by the Visit object.
+ */
+class VisitRequestProcessor extends RequestProcessor
+{
+    // TODO: much of the logic in this class should be moved to new service class
+
+    /**
+     * @var EventDispatcher
+     */
+    private $eventDispatcher;
+
+    /**
+     * @var VisitorRecognizer
+     */
+    private $visitorRecognizer;
+
+    /**
+     * @var Settings
+     */
+    private $userSettings;
+
+    /**
+     * @var int
+     */
+    private $visitStandardLength;
+
+    public function __construct(EventDispatcher $eventDispatcher, VisitorRecognizer $visitorRecognizer, Settings $userSettings,
+                                $visitStandardLength)
+    {
+        $this->eventDispatcher = $eventDispatcher;
+        $this->visitorRecognizer = $visitorRecognizer;
+        $this->userSettings = $userSettings;
+        $this->visitStandardLength = $visitStandardLength;
+    }
+
+    public function processRequestParams(VisitProperties $visitProperties, Request $request)
+    {
+        // the IP is needed by isExcluded() and GoalManager->recordGoals()
+        $visitProperties->visitorInfo['location_ip'] = $request->getIp();
+
+        // TODO: move VisitExcluded logic to here (or move to service class stored in DI)
+        $excluded = new VisitExcluded($request, $visitProperties->visitorInfo['location_ip']);
+        if ($excluded->isExcluded()) {
+            return true;
+        }
+
+        // visitor recognition
+        $visitorId = $this->userSettings->getConfigId($request, $visitProperties->visitorInfo['location_ip']);
+        $visitProperties->setRequestMetadata('CoreHome', 'visitorId', $visitorId);
```

Comment:
Those kinda things, when the core knows about it anyway, should be `$visitor->getId()`, `$visitor->setId()`, `$visitor->s/getIsVisitorKnown()` , `$visitor->isNewVisit()`, ...